emu_mps/custom_callbacks.py

```diff
@@ -0,0 +1,29 @@
+from emu_mps.mps import MPS
+import torch
+
+
+def calculate_entanglement_entropy(mps: MPS, b: int) -> float:
+    """
+    Calculate the von Neumann entanglement entropy at the bond between sites b-1 and b
+    S_E = -Tr(s^2 * log(s^2)), where s are the singular values at the chosen bond
+    """
+    # orthogonalize the MPS at site b
+    mps.orthogonalize(b)
+
+    # perform svd on reshaped tensor at index 'b' of the mps
+    tensor_at_site_b = mps.factors[b]
+    matrix = tensor_at_site_b.reshape(
```

Comment:
a shorter alternative using parameters to [`flatten`](https://pytorch.org/docs/stable/generated/torch.flatten.html)
`matrix = mps.factors[b].flatten(end_dim=1)`

Reply:
I let you and the other choose whatever you prefer, it was just a suggestion :) .
My opinion is that if there is a specialized torch function for sth, better use it.